dashboard/app/controllers/api/v1/pd/applications_controller.rb

```diff
@@ -44,6 +44,8 @@ def index
 
     # GET /api/v1/pd/applications/1
     def show
+      return head :forbidden if @application.status == 'incomplete' && !current_user.workshop_admin?
```

Comment:
I think this line and the follow lines that are similar should use `can?` or `authorize!`.  I think it would be good to try
```
authorize! show @application if @application.is_a? Pd::Application::TeacherApplication
```
This will get us to relying on CanCan to make the call if this should be forbidden and keep the logic in one place. 

We might be able to take this one step farther. By adding a `authorize_resource` call at the top of the controller which is specific two these 3 actions and applications that are teacher applications. However that seems like a nice to have so lets first get the first suggestion working if we can and then try for that if we feel there is time.

Reply:
Oo good ideas will explore!

Reply:
Hurray! Looks like I can get away with one `authorize_resource Pd::Application::TeacherApplication if @application.is_a? Pd::Application::TeacherApplication` at the top! Just added it